crates/red_knot_python_semantic/src/program.rs

```diff
@@ -144,9 +144,23 @@ pub enum PythonPath {
     /// [`sys.prefix`]: https://docs.python.org/3/library/sys.html#sys.prefix
     SysPrefix(SystemPathBuf),
 
+    /// An environment was active e.g. via `VIRTUAL_ENV`
+    ActiveEnvironment(SystemPathBuf),
+
     /// Resolved site packages paths.
     ///
     /// This variant is mainly intended for testing where we want to skip resolving `site-packages`
     /// because it would unnecessarily complicate the test setup.
     KnownSitePackages(Vec<SystemPathBuf>),
 }
+
+impl PythonPath {
+    pub fn find_virtual_env() -> Option<Self> {
+        let virtual_env = std::env::var("VIRTUAL_ENV").ok();
+        if let Some(virtual_env) = virtual_env {
+            tracing::debug!("Found virtual environment at {:?}", virtual_env);
+            return Some(Self::ActiveEnvironment(SystemPathBuf::from(virtual_env)));
+        }
+        None
+    }
+}
```

Comment:
Using this approach for when `VIRTUAL_ENV` is set does make sense to me because we probably want Red Knot to fail if *whatever `VIRTUAL_ENV`* is pointing to doesn't turn out to be a virtual env. 

I'm not sure if it's the right approach to e.g. support `.venv` because we then only want to use the folder if it actually is a virtual env.

Reply:
I should probably look at this for parity with uv..